custom_components/tuya_local/devices/cleanair_604b_humidifier.yaml

```diff
@@ -0,0 +1,165 @@
+name: Humidifier
+products:
+  - id: cqoisddoci9i3yer
+    name: Clean Air Optima CA-604B Smart Top Filling
+primary_entity:
+  entity: humidifier
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+      mapping:
+        - dps_val: false
+          icon: mdi:air-humidifier-off
```

Comment:
It looks as though HA is going to deprecate setting dynamic icons programmatically in favour of the new icon translations feature coming in HA 2024.2, so I prefer to keep the icons simple for now (just a static icon at the entity level where the default icon is not specific enough - usually for the generic entities like number, select, switch etc).